### PR TITLE
API: searching for syslog message

### PR DESCRIPTION
**Pull request by baroninn**
searching for syslog message ;-)

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Review by murrant — CHANGES_REQUESTED** [review]
This allows SQL injection.  Need to use parameters.

**Comment by baroninn** [comment]
@murrant
i'm no SQL injection specialist,,,, but i'm just doing the same as in other functions.. for example in "list_devices" function LINE: 307
how is my approach different ?
again: i'm just a novice programmer...
maybe you can help me understand what I need to change to prevent the SQL injection ?
since this is a very much needed functionality for my operations ;-)

thanks.

**Comment by nightcore500** [comment]
@baroninn 
You embed the content of the variable "msg" directly to the "query" variable. This allows SQL injections.
Use the placeholder "?" instead of the "msg" variable. (See also "prepared statements")

Like this (I have not checked the SQL query for correctness):
```
    $msg = $request->get('msg');
    if ($msg) {
        $query .= ' AND `msg` LIKE ?';
        $param[] = $msg;
    }
```

File: includes/html/api_functions.inc.php [hunk 1]
```diff
@@ -2319,6 +2319,12 @@ function list_logs(Illuminate\Http\Request $request, Router $router)
         $param[] = $to;
     }
 
+    $msg = $request->get('msg');
+    if ($msg) {
+        $query .= " AND `msg` LIKE '%" . $msg . "%'";
```
Comment (murrant) [inline includes/html/api_functions.inc.php:2324]: The $param part below is correct, just use this for the query.
``` $query .= " AND `msg` LIKE '%?%'";```

This also needs a small documentation update so users know it exists.

**Comment by murrant** [comment]
Closing due to inactivity, feel free to reopen.

**Pull request closed without merging**